package/MDAnalysis/analysis/density.py

```diff
@@ -497,12 +495,70 @@ def __repr__(self):
             grid_type = 'histogram'
         return '<Density ' + grid_type + ' with ' + str(self.grid.shape) + ' bins>'
 
+
+def _set_user_grid(gridcenter, xdim, ydim, zdim, smin, smax):
+    """Helper function to set the grid dimensions to user defined values
+
+    Parameters
+    ----------
+    gridcenter : numpy ndarray, float32
+            3 element ndarray containing the x, y and z coordinates of the grid
+            box center
+    xdim : float
+            Box edge length in the x dimension
+    ydim : float
+            Box edge length in the y dimension
+    zdim : float
+            Box edge length in the y dimension
+    smin : numpy ndarray, float32
+            Minimum x,y,z coordinates for the input selection
+    smax : numpy ndarray, float32
+            Maximum x,y,z coordinates for the input selection
+
+    Returns
+    -------
+    umin : numpy ndarray, float32
+            Minimum x,y,z coordinates of the user defined grid
+    umax : numpy ndarray, float32
+            Maximum x,y,z coordinates of the user defined grid
+    """
+    # Check user inputs
+    if (len(gridcenter) != 3) or not isinstance(gridcenter, np.ndarray):
```

Comment:
We can allow `gridcenter` any sequence. Replace 526 with
```python
gridcenter = np.asarray(gridcenter, dtype=np.float32)
if gridcenter.shape != (3,):
    raise ValueError("gridcenter must be a 3D coordinate")
try:
    xyzdim = np.array([xdim, ydim, zdim], dtype=np.float32)
except TypeError:
    raise TypeError("xdim, ydim, zdim must be numbers")
```

Reply:
Addressed in 248e459